fdbserver/storageserver.actor.cpp

```diff
@@ -8567,13 +8569,24 @@ ACTOR Future<Void> fetchKeys(StorageServer* data, AddingShard* shard) {
 					metricReporter.addFetchedBytes(expectedBlockSize, this_block.size());
 					totalBytes += expectedBlockSize;
 
+					if (!data->fetchKeysLimiter.ready().isReady()) {
```

Comment:
I wonder if this is just an enhanced version for `fetchKeysBudgetUsed`. What this is doing is similar to the existing `STORAGE_FETCH_KEYS_DELAY` mechanism. Can we consolidate these two?

Reply:
The difference is `fetchKeysBudgetUsed` is calculated per `commit`, without considering the actual time. If the new limiter works, we can unify them, but at this moment, I would prefer not changing the current code.